staging/src/k8s.io/component-base/cli/flag/sectioned.go

```diff
@@ -58,6 +58,17 @@ func (nfs *NamedFlagSets) FlagSet(name string) *pflag.FlagSet {
 	return nfs.FlagSets[name]
 }
 
+// MergeNamedFlagSets merges the given NamedFlagSets into a new NamedFlagSets instance with all flags combined.
+func MergeNamedFlagSets(nfss ...NamedFlagSets) NamedFlagSets {
+	newNFS := NamedFlagSets{}
+	for _, nfs := range nfss {
+		for _, flagName := range nfs.Order {
+			*newNFS.FlagSet(flagName) = *nfs.FlagSet(flagName)
```

Comment:
I'm wondering about preventing additional flagsets from overwriting the default flagsets.  It seems like `AddFlagSet` already protects against this.  Can you use this here?

Maybe something like 
```
newNFS.FlagSet(flagName).AddFlagSet(nfs.FlagSet(flagName))
```

Reply:
Good point, done.